include/swift/AST/DiagnosticsSema.def

```diff
@@ -1506,6 +1506,11 @@ ERROR(protocol_composition_not_protocol,none,
       "non-protocol type %0 cannot be used within a protocol composition", (Type))
 ERROR(objc_protocol_inherits_non_objc_protocol,none,
       "@objc protocol %0 cannot refine non-@objc protocol %1", (Type, Type))
+WARNING(protocol_composition_with_postfix,none,
+        "protocol composition with postfix '%0' is ambiguous "
+        "and will be rejected in future version of Swift", (StringRef))
+NOTE(protocol_composition_with_postfix_suppress_warning,none,
+     "wrap the composition in parentheses to suppress this warning", ())
```

Comment:
Oops, one other nitpick: we've historically used "silence this warning" instead of "suppress this warning". In this case, though, it's not even really about silencing the warning; it's the correct way to write the type. (And there's not really an alternate interpretation.) Maybe we should just put the fix-it on the warning?